app/infrastructure/services/tsa_service.py

```
@@ -0,0 +1,68 @@
+
+    async def verify(self, anchor_id: str) -> bool:
+        """Verify stored TSA token against payload_hash; update verification_status. Return True if valid."""
+        anchor = await self._repo.get_by_id(anchor_id)
+        if not anchor:
+            return False
+        digest = self._tsa_client.digest_for_chain_tip(anchor.payload_hash)
+        ok = self._tsa_client.verify(anchor.tsa_token, digest)
+        await self._repo.update_verification_status(
+            anchor_id,
+            TsaVerificationStatus.VERIFIED if ok else TsaVerificationStatus.FAILED,
+            verified_at=datetime.now(timezone.utc),
```

Ruff (0.15.5)

[warning] Use `datetime.UTC` alias

Convert to `datetime.UTC` alias

(UP017)